src/components/Appbar.tsx

```diff
@@ -15,20 +15,47 @@ import { ThemeToggler } from './ThemeToggler';
 import { NavigationMenu } from './landing/appbar/nav-menu';
 import SearchBar from './search/SearchBar';
 import MobileScreenSearch from './search/MobileScreenSearch';
+import { useCallback, useEffect, useState } from 'react';
 
 export const Appbar = () => {
   const session = useSession();
   const [sidebarOpen, setSidebarOpen] = useRecoilState(sidebarOpenAtom);
+  const [show, setShow] = useState(true);
+  const [lastScrollY, setLastScrollY] = useState(0);
   const currentPath = usePathname();
   const params = useParams();
   let bookmarkPageUrl = null;
   if (params.courseId && params.courseId[0]) {
     bookmarkPageUrl = `/courses/${params.courseId[0]}/bookmarks`;
   }
 
+  const controlAppbar = useCallback(() => {
+    if (typeof window !== 'undefined') {
+      if (window.scrollY > lastScrollY) { // if scroll down hide the navbar
+        setShow(false);
+      } else { // if scroll up show the navbar
+        setShow(true);
+      }
+
+      // remember current page location to use in the next move
+      setLastScrollY(window.scrollY);
```

Comment:
How is this not causing an infinite loop?
You're updating the `lastScrollY` here and have it as a dependency to the effect?
Is it because it's the same the second time?
Doesn't matter but we should confirm

Reply:
yes you are correct it is causing an issue. Actually I had added the useCallback after I was done testing the change. And then forgot to test again. Will correct and update. Sorry for the inconvinience.

Reply:
Still dont see a useCallback here?
This will cause the effect to run on every render

Reply:
but using useCallback was causing infinite loop and it was not rendering that's why I removed that.